conf/logging.go

```diff
@@ -2,59 +2,105 @@ package conf
 	File             string                 `mapstructure:"log_file" json:"log_file"`
 	DisableColors    bool                   `mapstructure:"disable_colors" split_words:"true" json:"disable_colors"`
 	QuoteEmptyFields bool                   `mapstructure:"quote_empty_fields" split_words:"true" json:"quote_empty_fields"`
 	TSFormat         string                 `mapstructure:"ts_format" json:"ts_format"`
 	Fields           map[string]interface{} `mapstructure:"fields" json:"fields"`
+	SQL              string                 `mapstructure:"sql" json:"sql"`
 }
 
-func ConfigureLogging(config *LoggingConfig) (*logrus.Entry, error) {
-	logger := logrus.New()
-
-	tsFormat := time.RFC3339Nano
-	if config.TSFormat != "" {
-		tsFormat = config.TSFormat
-	}
-	// always use the full timestamp
-	logger.SetFormatter(&logrus.TextFormatter{
-		FullTimestamp:    true,
-		DisableTimestamp: false,
-		TimestampFormat:  tsFormat,
-		DisableColors:    config.DisableColors,
-		QuoteEmptyFields: config.QuoteEmptyFields,
-	})
+func ConfigureLogging(config *LoggingConfig) error {
+	logrus.SetFormatter(&logrus.JSONFormatter{})
 
 	// use a file if you want
 	if config.File != "" {
-		f, errOpen := os.OpenFile(config.File, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0664)
-		if errOpen != nil {
-			return nil, errOpen
+		f, err := os.OpenFile(config.File, os.O_RDWR|os.O_APPEND|os.O_CREATE, 0664)
+		if err != nil {
+			return err
 		}
-		logger.SetOutput(f)
-		logger.Infof("Set output file to %s", config.File)
+		logrus.SetOutput(f)
+		logrus.Infof("Set output file to %s", config.File)
 	}
 
 	if config.Level != "" {
 		level, err := logrus.ParseLevel(config.Level)
 		if err != nil {
-			return nil, err
+			return err
 		}
-		logger.SetLevel(level)
-		logger.Debug("Set log level to: " + logger.GetLevel().String())
+		logrus.SetLevel(level)
+		logrus.Debug("Set log level to: " + logrus.GetLevel().String())
 	}
 
 	f := logrus.Fields{}
 	for k, v := range config.Fields {
 		f[k] = v
 	}
+	logrus.WithFields(f)
+
+	setPopLogger(config.SQL)
+
+	return nil
+}
+
+func setPopLogger(sql string) {
+	popLog := logrus.WithField("component", "pop")
+	sqlLog := logrus.WithField("component", "sql")
+
+	shouldLogSQL := sql == LOG_SQL_STATEMENT || sql == LOG_SQL_ALL
+	shouldLogSQLArgs := sql == LOG_SQL_ALL
 
-	return logger.WithFields(f), nil
+	pop.SetLogger(func(lvl plog.Level, s string, args ...interface{}) {
+		// Special case SQL logging since we have 2 extra flags to check
+		if lvl == logging.SQL {
+			if !shouldLogSQL {
+				return
+			}
+
+			if shouldLogSQLArgs && len(args) > 0 {
+				sqlLog.WithField("args", args).Info(s)
+			} else {
+				sqlLog.Info(s)
+			}
+			return
+		}
+
+		l := popLog
+		if len(args) > 0 {
+			l = l.WithField("args", args)
+		}
+
+		switch lvl {
+		case logging.Debug:
+			l.Debug(s)
+		case logging.Info:
+			l.Info(s)
+		case logging.Warn:
+			l.Warn(s)
+		case logging.Error:
+			l.Error(s)
+		}
+	})
```

Comment:
All of this argument parsing is handled by pop itself: https://github.com/gobuffalo/pop/blob/v5.3.4/logger.go#L56

I feel like this is too busy and adding too much configuration options. I would just add a new env var `GOTRUE_POP_LOG_LEVEL` and set that to equal the Pop levels.

Reply:
we can do that in another PR, will merge this in first so our logs are standardised